Client/UIWindowExtensions.swift

```diff
@@ -5,13 +5,18 @@
 import Foundation
 
 extension UIWindow {
-    
+    static var keyWindow: UIWindow? {
+        UIApplication.shared.windows.first(where: \.isKeyWindow)
+    }
     static var isLandscape: Bool {
-        return UIApplication.shared.windows
-            .first?
-            .windowScene?
-            .interfaceOrientation
+        interfaceOrientation?
             .isLandscape ?? false
     }
-    
+    static var isPortrait: Bool {
+        interfaceOrientation?
+            .isPortrait ?? false
+    }
+    static var interfaceOrientation: UIInterfaceOrientation? {
+        keyWindow?.windowScene?.interfaceOrientation
+    }
```

Comment:
It looks like UIApplication.shared.windows has a deprecation warning for iOS 15. Instead of this, have you looked at this approach?

```
static var keyWindow: UIWindow? {
        return UIApplication.shared.connectedScenes
            .filter { $0.activationState == .foregroundActive }
            .first(where: { $0 is UIWindowScene })
            .flatMap({ $0 as? UIWindowScene })?.windows
            .first(where: \.isKeyWindow)
    }
```

Maybe this can also achieve the same thing you're going for.

Reply:
nice catch. pushed as you suggested